src/sites/doc/components/demoblock/codeblock.tsx

```
@@ -4,26 +4,28 @@ import APPContext from '../../context'
-    <DemoBlock text={originCode} scss="">
-      <pre>
-        <code dangerouslySetInnerHTML={{ __html: highlightedCode }} />
-      </pre>
-    </DemoBlock>
-  )
+  try {
+    const highlightedCode = hljs.highlightAuto(originCode, ['jsx']).value
+    return (
+      <DemoBlock text={originCode} scss="">
+        <pre>
+          <code dangerouslySetInnerHTML={{ __html: highlightedCode }} />
```

Biome (1.9.4)

[error] Avoid passing content using the dangerouslySetInnerHTML prop.

Setting content using code can expose users to cross-site scripting (XSS) attacks

(lint/security/noDangerouslySetInnerHtml)